### PR TITLE
Fix write deny issue for S3 download

### PR DESCRIPTION
**Pull request by sunchunqiang**
1. Before download, set write permission 
2. add code to tap allow button  for ui test

### CODE REVIEW

**Comment by kvasukib** [comment]
@sunchunqiang Can you rebase your fork with the master and resolve the conflicts?

**Comment by kvasukib** [comment]
@sunchunqiang Can you test it against API levels 23 through 28?

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java [hunk 1]
```diff
@@ -357,12 +366,52 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         }
     }
 
+
+    private void  requestPermission()
```
Comment (kvasukib) [inline S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: remove extra whitespace and change style:

```
private void requestPermission() {
```

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java [hunk 2]
```diff
@@ -357,12 +366,52 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
+                    }
+                });
+
+                AlertDialog dialog = builder.create();
+                dialog.show();
+
+            } else {
+                makeRequest();
+            }
+        }
```
Comment (kvasukib) [inline S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: Add a else block and add a log statement saying that: 
```
Log.i(TAG, "Permission to store data in external storage is granted.");
```

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java [hunk 3]
```diff
@@ -357,12 +366,52 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
 
+
+    private void  requestPermission()
+    {
+        //ask for the permission in android M
+        int permission = ContextCompat.checkSelfPermission(this,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE);
+
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            Log.i(TAG, "Permission to record denied");
```
Comment (kvasukib) [inline S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: change to: "Permission to store data in external storage is not granted."

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java [hunk 4]
```diff
@@ -357,12 +366,52 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
+
+            if (ActivityCompat.shouldShowRequestPermissionRationale(this,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(this);
+                builder.setMessage("Permission to access the SD-CARD is required for this app to Download PDF.")
```
Comment (kvasukib) [inline S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: change message to: "Permission to access the External Storage is required for this application to store the downloaded data from Amazon S3"

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java [hunk 5]
```diff
@@ -336,6 +344,7 @@ public void onClick(View arg0) {
         });
 
         updateButtonAvailability();
+        requestPermission();
```
Comment (kvasukib) [inline S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: change function name to `requestWriteExternalStoragePermission()`

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java (hunk 1 above)
Comment (sunchunqiang) [reply S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: done

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java (hunk 2 above)
Comment (sunchunqiang) [reply S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: done

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java (hunk 3 above)
Comment (sunchunqiang) [reply S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: done !

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java (hunk 4 above)
Comment (sunchunqiang) [reply S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: done

File: S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java (hunk 5 above)
Comment (sunchunqiang) [reply S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/DownloadActivity.java]: done!

**Pull request merged**